django/db/models/fields/__init__.py

```diff
@@ -678,6 +675,12 @@ def contribute_to_class(self, cls, name, virtual_only=False):
             cls._meta.add_field(self, virtual=True)
         else:
             cls._meta.add_field(self)
+        if self.column:
+            # Refuse to override class methods with the descriptor. This leads to a situation
+            # where if you have a classmethod and a field similarly named, then such fields
+            # can't be deferred (we don't have a check for this yet).
+            if not getattr(cls, self.attname, None):
+                setattr(cls, self.attname, DeferredAttribute(self.attname, cls))
```

Comment:
Never mind, I forgot that the instance `__dict__` takes precedence over the descriptor. So this shouldn't have any negative impact on people not using defer/only. (It makes me wonder why the implementation of `DeferredAttribute.__get__` even bothers to check for the value in `instance.__dict__`, though -- if a value were found there, `DeferredAttribute.__get__` should never have even been called, right?)